plugins/search-backend-module-pg/src/database/DatabaseDocumentStore.ts

```diff
@@ -159,6 +167,10 @@ export class DatabaseDocumentStore implements DatabaseStore {
       query
         .from(tx.raw("documents, to_tsquery('english', ?) query", pgTerm))
         .whereRaw('query @@ body');
+
+      if (fallbackSearchQuery) {
+        query.or.whereRaw(fallbackSearchQuery);
```

Comment:
This will execute the fallbackSearchQuery every time, which might hurt performance. This could be better `SELECT document->>'title' FROM documents 
WHERE 
   CASE 
      WHEN to_tsquery('english', 'tarwar:*') @@ body THEN TRUE
      WHEN (custom_query) THEN TRUE
      ELSE FALSE 
   END;`